src/library-authoring/component-info/ComponentDetails.tsx

```diff
@@ -0,0 +1,47 @@
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Stack } from '@openedx/paragon';
+
+import { useLibraryBlockMetadata } from '../data/apiHooks';
+import HistoryWidget from '../generic/history-widget';
+import { ComponentDeveloperInfo } from './ComponentDeveloperInfo';
+import messages from './messages';
+
+interface ComponentDetailsProps {
+  usageKey: string;
+}
+
+const ComponentDetails = ({ usageKey }: ComponentDetailsProps) => {
+  const intl = useIntl();
+  const { data: componentMetadata } = useLibraryBlockMetadata(usageKey);
+
+  // istanbul ignore if: this should never happen
+  if (!componentMetadata) {
+    return null;
```

Comment:
I think we can throw an error instead of `null`. ([ex.](https://github.com/openedx/frontend-app-authoring/blob/master/src/library-authoring/create-collection/CreateCollectionModal.tsx#L23))

Reply:
Thank you @ChrisChV!
I added a better error handling here (a9a7b215f1349671c7e6739db7c1263155540eb2), adding a loading and error state, as this could have an error from api.